proto/protovalidate/buf/validate/validate.proto

```diff
@@ -3731,6 +3731,31 @@ message StringRules {
       }
     ];
 
+    // `ulid` specifies that the field value must be a valid ULID (Universally Unique
+    // Lexicographically Sortable Identifier) as defined by the [ULID specification](https://github.com/ulid/spec).
+    // ULID is a 26-character Base32-encoded string using Crockford's Base32 alphabet
+    // (0-9, A-Z excluding I, L, O, U). If the field value isn't a valid ULID, an error
+    // message will be generated.
+    //
+    // ```proto
+    // message MyString {
+    //   // value must be a valid ULID
+    //   string value = 1 [(buf.validate.field).string.ulid = true];
+    // }
+    // ```
+    bool ulid = 35 [
+      (predefined).cel = {
+        id: "string.ulid"
+        message: "value must be a valid ULID"
+        expression: "!rules.ulid || this == '' || this.matches('^[0-9A-HJKMNP-TV-Za-hjkmnp-tv-z]{26}$')"
```

Comment:
The first character of a ULID string cannot exceed 7. From the [spec](https://github.com/ulid/spec?tab=readme-ov-file#overflow-errors-when-parsing-base32-strings):

> Technically, a 26-character Base32 encoded string can contain 130 bits of information, whereas a ULID must only contain 128 bits. Therefore, the largest valid ULID encoded in Base32 is 7ZZZZZZZZZZZZZZZZZZZZZZZZZ, which corresponds to an epoch time of 281474976710655 or 2 ^ 48 - 1.

```suggestion
        expression: "!rules.ulid || this == '' || this.matches('^[0-7][0-9A-HJKMNP-TV-Za-hjkmnp-tv-z]{25}$')"
```